presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -219,47 +219,49 @@ protected RelationPlan visitJoin(Join node, Void context)
                 .build();
 
         ImmutableList.Builder<JoinNode.EquiJoinClause> equiClauses = ImmutableList.builder();
-        Expression postInnerJoinCriteria = new BooleanLiteral("TRUE");
+        List<Expression> nonComparisonExpressions = new ArrayList<>();
```

Comment:
I think you can remove this and just have the `postInnerJoinConditions` list, if you change the analyzer to only have special treatment for `=`